pkg/scheduler/scheduler.go

```diff
@@ -256,6 +264,47 @@ func findFlavorForResource(name corev1.ResourceName, val int64, wUsed map[string
 	return "", 0
 }
 
+func flavorSelector(spec *corev1.PodSpec, allowedKeys sets.String) nodeaffinity.RequiredNodeAffinity {
+	var specCopy corev1.PodSpec
```

Comment:
use `var requiredNodeAffinity RequiredNodeAffinity` directly?

Reply:
we need the spec because we are also collecting NodeSelector (see line 275 below). It is also the type we pass in the matching function at the end.

Reply:
I know. I means if you can put the NodeSelector and the Affinity you collected in RequiredNodeAffinity directly instead of using `nodeaffinity.GetRequiredNodeAffinity(&corev1.Pod{Spec: specCopy})` ?
```go
return RequiredNodeAffinity{labelSelector: labels.SelectorFromSet(nodeSelector), nodeSelector: affinity}
```

Reply:
we can't, labelSelector and nodeSelector are not exported